--- cocoa.app.mm	2025-07-01 09:59:38.249574500 +0800
+++ cocoa.app.1.mm	2025-07-07 15:53:06.392436600 +0800
@@ -84,5 +84,15 @@
     void application::quit() // NOLINT(*-static)
     {
         [NSApp stop:nil];
+        [NSApp postEvent:[NSEvent otherEventWithType:NSEventTypeApplicationDefined
+                                        location:NSZeroPoint
+                                   modifierFlags:0
+                                       timestamp:0
+                                    windowNumber:0
+                                         context:nil
+                                         subtype:0
+                                           data1:0
+                                           data2:0]
+             atStart:YES];
     }
 } // namespace saucer
